acl.go

```diff
@@ -34,120 +34,160 @@ type ACL struct {
-									goto out
+									goto unmatched
 								}
 							case "direction":
 								if cacheACL.Fields["direction"].(string) != value {
-									goto out
+									goto unmatched
 								}
 							case "match":
 								if cacheACL.Fields["match"].(string) != value {
-									goto out
+									goto unmatched
 								}
 							case "priority":
 								if cacheACL.Fields["priority"].(int) != value {
-									goto out
+									goto unmatched
 								}
 							case "log":
 								if cacheACL.Fields["log"].(bool) != value {
-									goto out
+									goto unmatched
 								}
 							case "external_ids":
 								if value == nil {
 									if len(cacheACL.Fields["external_ids"].(libovsdb.OvsMap).GoMap) != 0 {
-										goto out
+										goto unmatched
 									}
 								} else {
 									if !reflect.DeepEqual(cacheACL.Fields["external_ids"].(libovsdb.OvsMap).GoMap, value.(*libovsdb.OvsMap).GoMap) {
-										goto out
+										goto unmatched
 									}
 								}
 							}
 						}
 						return va.GoUUID, nil
-					out:
 					}
+				unmatched:
 				}
+				return "", ErrorNotFound
+			}
+		case libovsdb.UUID:
+			if va, ok := acls.(libovsdb.UUID); ok {
+				cacheACL, ok := odbi.cache[TableACL][va.GoUUID]
+				if !ok {
+					return "", ErrorSchema
+				}
+
+				for field, value := range row {
+					switch field {
+					case "action":
+						if cacheACL.Fields["action"].(string) != value {
+							goto out
+						}
+					case "direction":
+						if cacheACL.Fields["direction"].(string) != value {
+							goto out
+						}
+					case "match":
+						if cacheACL.Fields["match"].(string) != value {
+							goto out
+						}
+					case "priority":
+						if cacheACL.Fields["priority"].(int) != value {
+							goto out
+						}
+					case "log":
+						if cacheACL.Fields["log"].(bool) != value {
+							goto out
+						}
+					case "external_ids":
+						if value == nil {
+							if len(cacheACL.Fields["external_ids"].(libovsdb.OvsMap).GoMap) != 0 {
+								goto out
+							}
+						} else {
+							if !reflect.DeepEqual(cacheACL.Fields["external_ids"].(libovsdb.OvsMap).GoMap, value.(*libovsdb.OvsMap).GoMap) {
+								goto out
+							}
+						}
+					}
+				}
+				return va.GoUUID, nil
+			out:
 			}
 		}
 	}
 	return "", ErrorNotFound
 }
 
-func (odbi *ovndb) aclAddImp(lsw, direct, match, action string, priority int, external_ids map[string]string, logflag bool, meter string, severity string) (*OvnCommand, error) {
+func (odbi *ovndb) aclAddImp(entityType EntityType, entity, direct, match, action string, priority int, external_ids map[string]string, logflag bool, meter, severity string) (*OvnCommand, error) {
+	var table string
+	var condition []interface{}
+
+	switch entityType {
+	case LOGICAL_SWITCH:
+		condition = libovsdb.NewCondition("name", "==", entity)
+		table = TableLogicalSwitch
+	case PORT_GROUP:
+		condition = libovsdb.NewCondition("_uuid", "==", stringToGoUUID(entity))
```

Comment:
Why using UUID for port_group (while we use "name" for logical_switch). Is it better to use "name" so that it is consistent between the two entity types?

Reply:
Done.